pkg/proof/share_proof.go

```diff
@@ -0,0 +1,82 @@
+package proof
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/nmt"
+)
+
+// Validate runs basic validations on the proof then verifies if it is consistent.
```

Comment:
[question] seems copied from celestia-core: https://github.com/celestiaorg/celestia-core/blob/07c167e7eb35ec9fa11dde1fde40c56910d14d28/types/share_proof.go#L66-L133

Should we create a follow up to delete the copy of code in celestia-core?